### PR TITLE
detect missing() usage in function_argument_linter

### LINKED ISSUE #1546: function_argument_linter() could look for missing() handling and mention it in the lint message
I think the simplest fix for existing violations of `function_argument_linter()` is to set a default of `NULL` for functions violating the design principle.

This doesn't always align the functions to the intent of the design guide, but it is safer for functions which may cause breaking changes when changing the argument order, and from what I've seen it's usually a good choice.

One case I can think of where it's _not_ safe to blithely make this switch is when such arguments are handled with `missing()`. Consider:

```r
foo <- function(x, y = 2, z) {
  if (missing(z)) z <- 3
  x * y * z
}
```

And existing usage `foo(1, 2)`, which will break if changing to `z = NULL` by default -- there was an implicit default value of `z` set inside the function. We can of course accommodate this by changing `missing(z)` to `is.null(z)` or even making the default `z = 3` explicit, but it is more complicated, at least.

OTOH, it should be straightforward for the linter logic to look, when identifying parameter `zed` as violating `function_argument_linter()`, for usages of `missing(zed)` in the applicable function body, and if found, mention this caveat in the linter message.

PS I'm also curious if anyone can think of other edge cases where setting this default can break existing code.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #1546

### CODE REVIEW

**Comment by AshesITR** [comment]
Can we add a test case without `{}` in the function body?

```r
function(x = 2, y) if (missing(y)) x else y
```

Just to make sure the XPath works everywhere.
The rest looks good.

File: NEWS.md [hunk 1]
```diff
@@ -30,6 +30,8 @@
 * `conjunct_test_linter()` also lints usage like `dplyr::filter(x, A & B)` in favor of using `dplyr::filter(x, A, B)` (part of #884, @MichaelChirico).
 * `sort_linter()` checks for code like `x == sort(x)` which is better served by using the function `is.unsorted()` (part of #884, @MichaelChirico).
 * `paste_linter()` gains detection for file paths that are better constructed with `file.path()`, e.g. `paste0(dir, "/", file)` would be better as `file.path(dir, file)` (part of #884, @MichaelChirico).
+* `fixed_regex_linter()` gains an option `allow_unescaped` (default `FALSE`) to toggle linting regexes not requiring any escapes or character classes (#1689, @MichaelChirico). Thus `fixed_regex_linter(allow_unescaped = TRUE)` would lint on `grepl("[$]", x)` but not on `grepl("a", x)` since the latter does not use any regex special characters.
```
Comment (AshesITR) [inline NEWS.md:33]: Merge messed up?

**Review by AshesITR — DISMISSED** [review]

File: NEWS.md [hunk 2]
```diff
@@ -30,6 +30,7 @@
 * `conjunct_test_linter()` also lints usage like `dplyr::filter(x, A & B)` in favor of using `dplyr::filter(x, A, B)` (part of #884, @MichaelChirico).
 * `sort_linter()` checks for code like `x == sort(x)` which is better served by using the function `is.unsorted()` (part of #884, @MichaelChirico).
 * `paste_linter()` gains detection for file paths that are better constructed with `file.path()`, e.g. `paste0(dir, "/", file)` would be better as `file.path(dir, file)` (part of #884, @MichaelChirico).
+* `function_argument_linter()` detects usage of `missing()` for the linted argument (#1546, @MichaelChirico). The simplest fix for `function_argument_linter()` lints is typically to set that argument to `NULL` by default, in which case it's usually preferable to update function logic checking `missing()` to check `is.null()` instead.
```
Comment (MichaelChirico) [inline NEWS.md:33]: lol this zombie NEWS item refuses to die

**Comment by MichaelChirico** [comment]
Oh, whoops. Should open my eyes wider. I mis-read and thought it was the `fixed_regex_linter()` bullet back again. Restored.

**Review by AshesITR — APPROVED** [review]
